src/App.tsx

```diff
@@ -1,19 +1,141 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const preparedTodos = todos.map((todo) => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null,
+  };
+});
+
+type State = {
+  todosList: Todo[],
+  taskText: string,
+  selectedUser: string,
+  hasUserSelectedMessage: boolean,
+  titleIsEmpty: boolean,
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todosList: [...preparedTodos],
+    taskText: '',
+    selectedUser: '',
+    hasUserSelectedMessage: false,
+    titleIsEmpty: false,
+  };
+
+  clearForm = () => {
+    this.setState({ taskText: '', selectedUser: '' });
+  };
+
+  validateForm = () => {
+    this.setState(state => ({ hasUserSelectedMessage: state.taskText === '' }));
+    this.setState(state => ({ titleIsEmpty: state.taskText === '' }));
+  };
+
+  handleTitleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      taskText: event.target.value,
+      titleIsEmpty: false,
+    });
+  };
+
+  handleSelectField = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      selectedUser: event.target.value,
+      hasUserSelectedMessage: false,
+    });
+  };
+
+  addTodo = () => {
+    const { selectedUser } = this.state;
+
+    const currentUser: User | null
+      = users.find((user) => user.name === selectedUser) || null;
+
+    if (!currentUser) {
+      return;
+    }
+
+    this.setState((state) => ({
+      todosList: [...state.todosList, {
+        userId: 1,
```

Comment:
why you hardcode userId?

Reply:
I made it because in this task userID is the same everywhere, but I know that I could have used currentUser.id instead.